rpc/tendermint/client/client.go

```diff
@@ -159,36 +94,68 @@ func CallCode(client rpcclient.Client,
 
 func Call(client rpcclient.Client, fromAddress, toAddress,
 	data []byte) (*rpc_types.ResultCall, error) {
-	var res rpc_types.ErisDBResult
-	var err error
-	switch cli := client.(type) {
-	case *rpcclient.ClientJSONRPC:
-		_, err = cli.Call("call", []interface{}{fromAddress, toAddress, data}, &res)
-	case *rpcclient.ClientURI:
-		_, err = cli.Call("call", map[string]interface{}{
-			"fromAddress": fromAddress,
-			"toAddress":   toAddress,
-			"data":        data,
-		}, &res)
-
-	}
+	res, err := performCall(client, "call",
+		"fromAddress", fromAddress,
+		"toAddress", toAddress,
+		"data", data)
 	if err != nil {
 		return nil, err
 	}
 	return res.(*rpc_types.ResultCall), err
 }
 
 func GetName(client rpcclient.Client, name string) (*core_types.NameRegEntry, error) {
-	var res rpc_types.ErisDBResult
-	var err error
-	switch cli := client.(type) {
-	case *rpcclient.ClientJSONRPC:
-		_, err = cli.Call("get_name", []interface{}{name}, &res)
-	case *rpcclient.ClientURI:
-		_, err = cli.Call("get_name", map[string]interface{}{"name": name}, &res)
-	}
+	res, err := performCall(client, "get_name",
+		"name", name)
 	if err != nil {
 		return nil, err
 	}
 	return res.(*rpc_types.ResultGetName).Entry, nil
 }
+
+func BlockchainInfo(client rpcclient.Client, minHeight,
+	maxHeight int) (*rpc_types.ResultBlockchainInfo, error) {
+	res, err := performCall(client, "blockchain",
+		"minHeight", minHeight,
+		"maxHeight", maxHeight)
+	if err != nil {
+		return nil, err
+	}
+	return res.(*rpc_types.ResultBlockchainInfo), err
+}
+
+func performCall(client rpcclient.Client, method string,
+	paramKeyVals ...interface{}) (res rpc_types.ErisDBResult, err error) {
+	paramsMap, paramsSlice := mapAndValues(paramKeyVals...)
+	switch cli := client.(type) {
+	case *rpcclient.ClientJSONRPC:
+		_, err = cli.Call(method, paramsSlice, &res)
+	case *rpcclient.ClientURI:
+		_, err = cli.Call(method, paramsMap, &res)
+	}
+	return res, err
+
+}
+func mapAndValues(orderedKeyVals ...interface{}) (map[string]interface{}, []interface{}) {
+	if len(orderedKeyVals)%2 != 0 {
+		panic(fmt.Sprintf("mapAndValues requires a even length list of keys and "+
```

Comment:
Yeah I guess I could have just let it panic array out of bounds, so I saw this as making the panic nicer. But I agree.